libs/state/src/lib/rx-state.ts

```diff
@@ -46,30 +69,51 @@ export type RxStateSetupFn<State extends object> = (
  * @docsPage RxState
  *
  */
+export function rxState<State extends object>(): RxState<State>;
+export function rxState<State extends object>(
+  setupFn: RxStateSetupFn<State>,
+): RxState<State>;
 export function rxState<State extends object>(
-  setupFn?: RxStateSetupFn<State>,
+  options: RxStateOptions,
+): RxState<State>;
+export function rxState<State extends object>(
+  setupFn: RxStateSetupFn<State>,
+  options: RxStateOptions,
+): RxState<State>;
+export function rxState<State extends object>(
+  setupFnOrOptions?: RxStateSetupFn<State> | RxStateOptions,
+  options?: RxStateOptions,
 ): RxState<State> {
-  assertInInjectionContext(rxState);
+  const injectorFromOptions = getInjectorFromOptions(setupFnOrOptions, options);
+  if (!injectorFromOptions) {
+    assertInInjectionContext(rxState);
+  }
+
+  const injector = injectorFromOptions ?? inject(Injector);
 
-  const legacyState = new LegacyState<State>();
-  const destroyRef = inject(DestroyRef);
+  return runInInjectionContext(injector, () => {
+    const legacyState = new LegacyState<State>();
+    const destroyRef = inject(DestroyRef);
 
-  destroyRef.onDestroy(() => legacyState.ngOnDestroy());
+    destroyRef.onDestroy(() => legacyState.ngOnDestroy());
 
-  const state: RxState<State> = {
-    get: legacyState.get.bind(legacyState),
-    set: legacyState.set.bind(legacyState),
-    connect: legacyState.connect.bind(legacyState),
-    select: legacyState.select.bind(legacyState),
-    signal: legacyState.signal.bind(legacyState),
-    computed: legacyState.computed.bind(legacyState),
-    computedFrom: legacyState.computedFrom.bind(legacyState),
-    $: legacyState.$,
-    setAccumulator: legacyState.setAccumulator.bind(legacyState),
-    asReadOnly: legacyState.asReadOnly.bind(legacyState),
-  };
+    const state: RxState<State> = {
+      get: legacyState.get.bind(legacyState),
+      set: legacyState.set.bind(legacyState),
+      connect: legacyState.connect.bind(legacyState),
+      select: legacyState.select.bind(legacyState),
+      signal: legacyState.signal.bind(legacyState),
+      computed: legacyState.computed.bind(legacyState),
+      computedFrom: legacyState.computedFrom.bind(legacyState),
+      $: legacyState.$,
+      setAccumulator: legacyState.setAccumulator.bind(legacyState),
+      asReadOnly: legacyState.asReadOnly.bind(legacyState),
+    };
 
-  setupFn?.(state);
+    if (setupFnOrOptions && typeof setupFnOrOptions === 'function') {
+      setupFnOrOptions?.(state);
+    }
```

Comment:
Could be simplified:

```suggestion
    if (typeof setupFnOrOptions === 'function') {
      setupFnOrOptions(state);
    }
```

Reply:
Done